sdks/go/pkg/beam/io/fhirio/utils_test.go

```diff
@@ -0,0 +1,105 @@
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fhirio
+
+import (
+	"errors"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"net/http"
+	"testing"
+)
+
+var (
+	fakeRequestReturnErrorMessage = "internal error"
+	requestReturnErrorFakeClient  = &fakeFhirStoreClient{
+		fakeReadResources: func(resource string) (*http.Response, error) {
+			return nil, errors.New(fakeRequestReturnErrorMessage)
+		},
+		fakeExecuteBundles: func(storePath string, bundle []byte) (*http.Response, error) {
+			return nil, errors.New(fakeRequestReturnErrorMessage)
+		},
+	}
+
+	fakeBadStatus         = "403 Forbidden"
+	badStatusFakeResponse = &http.Response{Status: fakeBadStatus}
+	badStatusFakeClient   = &fakeFhirStoreClient{
+		fakeReadResources: func(resource string) (*http.Response, error) {
+			return badStatusFakeResponse, nil
+		},
+		fakeExecuteBundles: func(storePath string, bundle []byte) (*http.Response, error) {
+			return badStatusFakeResponse, nil
+		},
+	}
+
+	fakeBodyReaderErrorMessage  = "ReadAll fail"
+	bodyReaderErrorFakeResponse = &http.Response{
+		Body: &fakeReaderCloser{
+			fakeRead: func([]byte) (int, error) {
+				return 0, errors.New(fakeBodyReaderErrorMessage)
+			},
+		}, Status: "200 Ok"}
+	bodyReaderErrorFakeClient = &fakeFhirStoreClient{
+		fakeReadResources: func(resource string) (*http.Response, error) {
+			return bodyReaderErrorFakeResponse, nil
+		},
+		fakeExecuteBundles: func(storePath string, bundle []byte) (*http.Response, error) {
+			return bodyReaderErrorFakeResponse, nil
+		},
+	}
+)
+
+type fakeFhirStoreClient struct {
+	fakeReadResources  func(string) (*http.Response, error)
+	fakeExecuteBundles func(storePath string, bundle []byte) (*http.Response, error)
+}
+
+func (c *fakeFhirStoreClient) executeBundle(storePath string, bundle []byte) (*http.Response, error) {
+	return c.fakeExecuteBundles(storePath, bundle)
+}
+
+func (c *fakeFhirStoreClient) readResource(resourcePath string) (*http.Response, error) {
+	return c.fakeReadResources(resourcePath)
+}
+
+// Useful to fake the Body of a http.Response.
+type fakeReaderCloser struct {
+	fakeRead func([]byte) (int, error)
+}
+
+func (*fakeReaderCloser) Close() error {
+	return nil
+}
+
+func (m *fakeReaderCloser) Read(b []byte) (int, error) {
+	return m.fakeRead(b)
+}
+
+func validateResourceErrorCounter(t *testing.T, pipelineResult beam.PipelineResult, expectedCount int) {
```

Comment:
This  method should not performa the t.Fatalf but return an error back to the testing function that can then check for error and display appropriate message within the tests context. Additionally if you were to keep  this helper util you would likely want to call t.Helper().